src/model/event.rs

```diff
@@ -2239,24 +2239,30 @@ macro_rules! with_related_ids_for_event_types {
                     Interaction::ApplicationCommand(i) => Some(i.user.id),
                     Interaction::MessageComponent(i) => Some(i.user.id),
                     Interaction::Autocomplete(i) => Some(i.user.id),
+                    Interaction::ModalSubmit(i) => Some(i.user.id),
                 },
                 guild_id: match &e.interaction {
                     Interaction::Ping(_) => None,
                     Interaction::ApplicationCommand(i) => i.guild_id.into(),
                     Interaction::MessageComponent(i) => i.guild_id.into(),
                     Interaction::Autocomplete(i) => i.guild_id.into(),
+                    Interaction::ModalSubmit(i) => i.guild_id.into(),
                 },
                 channel_id: match &e.interaction {
                     Interaction::Ping(_) => None,
                     Interaction::ApplicationCommand(i) => Some(i.channel_id),
                     Interaction::MessageComponent(i) => Some(i.channel_id),
                     Interaction::Autocomplete(i) => Some(i.channel_id),
+                    Interaction::ModalSubmit(i) => Some(i.channel_id),
                 },
                 message_id: match &e.interaction {
                     Interaction::Ping(_) => None,
                     Interaction::ApplicationCommand(_) => None,
                     Interaction::MessageComponent(i) => Some(i.message.id),
                     Interaction::Autocomplete(i) => None,
+                    Interaction::ModalSubmit(i) => {
+                        if let Option::Some(m) = &i.message { Some(m.id) } else { None }
+                    },
```

Comment:
```suggestion
                    Interaction::ModalSubmit(i) => i.message.as_ref().map(|m| m.id),
```

Reply:
map doesn't seem to work here. Since it expects a RelatedId and not Option.

Reply:
Ah, I forgot `RelatedId` borrowed `Option`'s variants; however, you can still use the `map` by converting the `Option` to `RelatedId` (`RelatedId` has a `From<Option<T>>` impl):
```suggestion
                    Interaction::ModalSubmit(i) => i.message.as_ref().map(|m| m.id).into(),
```